pkg/event/event.go

```diff
@@ -222,14 +262,27 @@ func NewInvocationEvent(opts NewInvocationEventOpts) Event {
 	}
 	evt.Name = InvokeFnName
 
+	correlationID := ""
+	if opts.CorrelationID != nil {
+		correlationID = *opts.CorrelationID
+	}
+
+	spanID := ""
+	if opts.SpanID != nil {
+		spanID = *opts.SpanID
+	}
+
 	evt.Data[consts.InngestEventDataPrefix] = InngestMetadata{
-		InvokeFnID: opts.FnID,
-		InvokeCorrelationId: func() string {
-			if opts.CorrelationID != nil {
-				return *opts.CorrelationID
-			}
-			return ""
-		}(),
+		InvokeFnID:          opts.FnID,
+		InvokeCorrelationId: correlationID,
+		InvokeSpanID:        spanID,
+		InvokeTimestamp:     opts.SpanTimestamp,
+		InvokeExpiresAt:     opts.ExpiresAt,
+		InvokeGroupID:       opts.GroupID,
+		InvokeDisplayName:   opts.DisplayName,
+		SourceAppID:         opts.SourceAppID,
+		SourceFnID:          opts.SourceFnID,
+		SourceFnVersion:     opts.SourceFnVersion,
```

Comment:
Makes perfect sense, though it's a shame we have to push so much data into the event, as it's visible to users and has mostly a tonne of concepts they don't otherwise have access to.